Subject: Replace unmaintained ansi-term with nu-ansi-term

Forwarded: not-needed
Comment: clap 4 doesn't use ansi-term or nu-ansi-term directly
----
diff --git a/Cargo.toml b/Cargo.toml
index bb6bf09..a0693a7 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -102,1 +102,1 @@
-color = ["ansi_term", "atty"]
+color = ["nu-ansi-term", "atty"]
@@ -112,2 +112,2 @@
-[target."cfg(not(windows))".dependencies.ansi_term]
-version = "0.12"
+[target."cfg(not(windows))".dependencies.nu-ansi-term]
+version = "0.50"
diff --git a/src/fmt.rs b/src/fmt.rs
index f447565..3078d2f 100644
--- a/src/fmt.rs
+++ b/src/fmt.rs
@@ -1,8 +1,8 @@
 #[cfg(all(feature = "color", not(target_os = "windows")))]
-use ansi_term::ANSIString;
+use nu_ansi_term::AnsiString;
 
 #[cfg(all(feature = "color", not(target_os = "windows")))]
-use ansi_term::Colour::{Green, Red, Yellow};
+use nu_ansi_term::Color::{Green, Red, Yellow};
 
 use std::env;
 use std::fmt;
@@ -129,12 +129,12 @@ pub enum Format<T> {
 
 #[cfg(all(feature = "color", not(target_os = "windows")))]
 impl<T: AsRef<str>> Format<T> {
-    fn format(&self) -> ANSIString {
+    fn format(&self) -> AnsiString {
         match *self {
             Format::Error(ref e) => Red.bold().paint(e.as_ref()),
             Format::Warning(ref e) => Yellow.paint(e.as_ref()),
             Format::Good(ref e) => Green.paint(e.as_ref()),
-            Format::None(ref e) => ANSIString::from(e.as_ref()),
+            Format::None(ref e) => AnsiString::from(e.as_ref()),
         }
     }
 }
@@ -169,8 +169,8 @@ impl<T: fmt::Display> fmt::Display for Format<T> {
 #[cfg(all(test, feature = "color", not(target_os = "windows")))]
 mod test {
     use super::Format;
-    use ansi_term::ANSIString;
-    use ansi_term::Colour::{Green, Red, Yellow};
+    use nu_ansi_term::AnsiString;
+    use nu_ansi_term::Color::{Green, Red, Yellow};
 
     #[test]
     fn colored_output() {
@@ -186,7 +186,7 @@ mod test {
         let none = Format::None("none");
         assert_eq!(
             &*format!("{}", none),
-            &*format!("{}", ANSIString::from("none"))
+            &*format!("{}", AnsiString::from("none"))
         );
     }
 }
diff --git a/src/lib.rs b/src/lib.rs
index 8c47b77..bfdd83c 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -541,7 +541,7 @@
 #![allow(bare_trait_objects, deprecated)]
 
 #[cfg(all(feature = "color", not(target_os = "windows")))]
-extern crate ansi_term;
+extern crate nu_ansi_term;
 #[cfg(feature = "color")]
 extern crate atty;
 #[macro_use]
